### PR TITLE
Collector table description for develop guide.

### PR DESCRIPTION
**Pull request by peng-yongsheng**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [X] New feature provided
- [ ] Improve performance

Register table section finish.

### CODE REVIEW

File: docs/en/Collector-Table-Description.md [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+### ApplicationComponent
+- Table name: application_component_`TimeDimension`
+- TimeDimension contains minute, hour, day, month
+- It is primarily used for the view of node type in application topology.
+
+Column Name | Short Name | Data Type | Description
+----------- | ---------- | --------- | ---------
+_id | _id | Keyword | primary key, es speciality, `time_bucket`_`metric_id`
+metric_id | c1 | Keyword | `application_id`_`component_id`
+component_id | c2 | Integer | [Component id](https://github.com/apache/incubator-skywalking/blob/master/apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/ComponentsDefine.java)
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: The link is very dangerous for document. Please use related path as same as other links in this document.

File: docs/en/Collector-Table-Description.md [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+application_id | c3 | Integer | Owner application id
+time_bucket | tb | Long | [Date format](Collector-Table-Description.md#Metric-table-time-bucket)
+
+### ApplicationMapping
+- Table name: application_mapping_`TimeDimension`
+- TimeDimension contains minute, hour, day, month
+- For example: A application invoke B application, collector will generate two metrics:
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: > A application invoke B application, collector will generate two metrics:

Application A calls Application B, collector generates the following two metrics:

File: docs/en/Collector-Table-Description.md [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+business_transaction_calls | b1 | Long | 
+business_transaction_error_calls | b2 | Long | 
+business_transaction_duration_sum | b3 | Long | 
+business_transaction_error_duration_sum | b4 | Long | 
+business_transaction_average_duration | b5 | Long | 
+mq_transaction_calls | m1 | Long | 
+mq_transaction_error_calls | m2 | Long | 
+mq_transaction_duration_sum | m3 | Long | 
+mq_transaction_error_duration_sum | m4 | Long | 
+mq_transaction_average_duration | m5 | Long |
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: Do you need to provide documents about business_transaction_* and mq_transaction_* ? Or still just for roadmap?

File: docs/en/Collector-Table-Description.md [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+business_transaction_error_calls | b2 | Long | 
+business_transaction_duration_sum | b3 | Long | 
+business_transaction_error_duration_sum | b4 | Long | 
+business_transaction_average_duration | b5 | Long | 
+mq_transaction_calls | m1 | Long | 
+mq_transaction_error_calls | m2 | Long | 
+mq_transaction_duration_sum | m3 | Long | 
+mq_transaction_error_duration_sum | m4 | Long | 
+mq_transaction_average_duration | m5 | Long | 
+satisfied_count | d1 | Long | [The formula](../../apm-collector/apm-collector-core/src/main/java/org/apache/skywalking/apm/collector/core/util/ApdexThresholdUtils.java)
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: Add `appdex` in description, `appdex` is a common concept in APM field. I think most people are easy to understand that.

File: docs/en/Collector-Table-Description.md [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+tolerating_count | d2 | Long | 
+frustrated_count | d3 | Long |
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: You should mention all these three fields are `appdex` related. Hard to know from this document

File: docs/en/Collector-Table-Description.md [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+time_bucket | tb | Long | [Date format](Collector-Table-Description.md#Metric-table-time-bucket)
+
+### ApplicationReferenceMetric
+- Table name: application_reference_metric_`TimeDimension`
+- TimeDimension contains minute, hour, day, month
+
+Column Name | Short Name | Data Type | Description
+----------- | ---------- | --------- | ---------
+_id | _id | Keyword | primary key, es speciality, `time_bucket`_`metric_id`
+metric_id | c1 | Keyword | `front_application_id`_`behind_application_id`_`source_value`
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: From the preview, add white space like this, which should be your expected.
> `front_application_id` _ `behind_application_id` _ `source_value`

File: docs/en/Collector-Table-Description.md [hunk 7]
```diff
@@ -0,0 +1,572 @@
...
+business_transaction_calls | b1 | Long | 
+business_transaction_error_calls | b2 | Long | 
+business_transaction_duration_sum | b3 | Long | 
+business_transaction_error_duration_sum | b4 | Long | 
+business_transaction_average_duration | b5 | Long | 
+mq_transaction_calls | m1 | Long | 
+mq_transaction_error_calls | m2 | Long | 
+mq_transaction_duration_sum | m3 | Long | 
+mq_transaction_error_duration_sum | m4 | Long | 
+mq_transaction_average_duration | m5 | Long |
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: Same concern I mentioned.

File: docs/en/Collector-Table-Description.md [hunk 8]
```diff
@@ -0,0 +1,572 @@
...
+business_transaction_error_duration_sum | b4 | Long | 
+business_transaction_average_duration | b5 | Long | 
+mq_transaction_calls | m1 | Long | 
+mq_transaction_error_calls | m2 | Long | 
+mq_transaction_duration_sum | m3 | Long | 
+mq_transaction_error_duration_sum | m4 | Long | 
+mq_transaction_average_duration | m5 | Long | 
+satisfied_count | d1 | Long | [The formula](../../apm-collector/apm-collector-core/src/main/java/org/apache/skywalking/apm/collector/core/util/ApdexThresholdUtils.java)
+tolerating_count | d2 | Long | 
+frustrated_count | d3 | Long |
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: Same concern I mentioned.

File: docs/en/Collector-Table-Description.md [hunk 9]
```diff
@@ -0,0 +1,572 @@
...
+### GlobalTrace
+- Table name: global_trace
+- The relationship between trace id and segment id is many to many.
+
+Column Name | Short Name | Data Type | Description
+----------- | ---------- | --------- | ---------
+_id | _id | Keyword | primary key, es speciality, `global_trace_id`_`segment_id`
+segment_id | c1 | Keyword | 
+global_trace_id | c2 | Keyword | 
+time_bucket | tb | Long | Second date format, [Date format](Collector-Table-Description.md#Metric-table-time-bucket)
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: What do you mean about `Second date format`? It is a confused naming.

File: docs/en/Collector-Table-Description.md [hunk 10]
```diff
@@ -0,0 +1,572 @@
...
+- The relationship between trace id and segment id is many to many.
+
+Column Name | Short Name | Data Type | Description
+----------- | ---------- | --------- | ---------
+_id | _id | Keyword | primary key, es speciality, `global_trace_id`_`segment_id`
+segment_id | c1 | Keyword | 
+trace_id | c2 | Keyword | 
+application_id | c3 | Integer | Owner of the segment
+service_name | c4 | Text | The entry span's operation name in this segment
+duration | c5 | Long | The cost duration of this segment
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: `cost duration` should be named as `duration` only, or `time span`?

File: docs/en/Collector-Table-Description.md [hunk 11]
```diff
@@ -0,0 +1,572 @@
...
+### GCMetric
+- Table name: gc_metric_`TimeDimension`
+- TimeDimension contains second, minute, hour, day, month
+
+Column Name | Short Name | Data Type | Description
+----------- | ---------- | --------- | ---------
+_id | _id | Keyword | primary key, es speciality, the value is `time_bucket`_`metric_id`
+metric_id | c1 | Keyword | the value is `instance_id`_`phrase`
+instance_id | c2 | Integer | Owner instance id
+phrase | c3 | Integer | [GCPhrase](https://github.com/apache/incubator-skywalking-data-collect-protocol/blob/master/JVMMetricsService.proto#L80-L83)
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: Related link path please.

File: docs/en/Collector-Table-Description.md [hunk 12]
```diff
@@ -0,0 +1,572 @@
...
+count | c4 | Long | GC count, sums values aggregate by `time_bucket`
+times | c5 | Long | The records received times in this time bucket
+time_bucket | tb | Long | [A formatted date](Collector-Table-Description.md#Metric-table-time-bucket)
```
Comment (wu-sheng) [inline docs/en/Collector-Table-Description.md]: This is a different name of `Date format`. Which one do you prefer?

Comment (wu-sheng) [reply docs/en/Collector-Table-Description.md]: Look like you change the naming way from this section, why?

File: docs/en/Collector-Table-Description.md (hunk 3 above)
Comment (peng-yongsheng) [reply docs/en/Collector-Table-Description.md]: just roadmap, never used at now.

**Comment by wu-sheng** [comment]
@peng-yongsheng  is this pr still in beta roadmap? Look like have no update more than 20 days.

**Comment by wu-sheng** [comment]
Please add link into the doc readme.

**Pull request merged**